tests/shadow-global-constructor.egg

```diff
@@ -0,0 +1,4 @@
+(datatype MySort
+  (Wrap i64))
+
+(let Wrap 3)
```

Comment:
lol this is a fun one! 😬

Reply:
I assume we want this one and the next one to fail ideally? seems a bit worrisome to allow this kind of shadowing if we disallow the other kind